Update contributors on OSCAR website (#344)
oscar-system/oscar-website: contributors.md

    email: thomas.breuer@math.rwth-aachen.de
    github: ThomasBreuer
    paid_by_dfg: true

  - name: Taylor Brysiewicz
    affiliation: University of Western Ontario
    website: https://sites.google.com/view/taylorbrysiewicz/home
    email: tbrysiew@uwo.ca
    github: tbrysiewicz

  - name: Tobias Boege
    affiliation: KTH Stockholm
    website: https://taboege.de/
    email: post@taboege.de
    github: taboege

  - name: Janko Böhm
    affiliation: University of Kaiserslautern-Landau
    website: https://agag-jboehm.math.rptu.de/~boehm/
    email: boehm@mathematik.uni-kl.de
    github: jankoboehm

  - name: Tobias Braun
    affiliation: RWTH Aachen
    website: https://www.math.rwth-aachen.de/~Tobias.Braun/
    email: tobias.braun2@rwth aachen
    github: SirToby25

  - name: Andrei Comaneci
    github: andreicomaneci

  - name: Antony Della Vecchia
    affiliation: TU Berlin
    website: https://antonydellavecchia.github.io/
    email: vecchia@math.tu-berlin.de
    github: antonydellavecchia

  - name: Alexander Demin
    affiliation: HSE University
    website: https://sumiya11.github.io/
    email: asdemin_2@edu.hse.ru
    github: sumiya11

  - name: Berenike Dieterle

>     github: BD-00

Reviewer: ```suggestion
    affiliation: University of Kaiserslautern-Landau
    github: BD-00
```

    github: lgoettgens

  - name: Fabian Gundlach
    github: fagu

  - name: Jeroen Hanselmann
    affiliation: University of Kaiserslautern-Landau
    website: https://hanselmanj.eu/
    email: hanselman@mathematik.uni-kl.de
    github: JHanselman

  - name: Alexej Jordan
    affiliation: TU Berlin
    website: https://www.tu.berlin/dmg/ueber-uns/team
    email: jordan@math.tu-berlin.de
    github: alexej-jordan

  - name: Lars Kastner
    affiliation: TU Berlin
    website: https://lkastner.github.io/
    email: kastner@math.tu-berlin.de
    github: lkastner

  - name: Aaruni Kaushik
    affiliation: University of Kaiserslautern-Landau
    website: https://akaushik.edufor.me
    email: akaushik@mathematik.uni-kl.de
    github: aaruni96

  - name: Lukas Kühne
    affiliation: Bielefeld University
    website: https://www.math.uni-bielefeld.de/~lkuehne/
    email: lukas.kuehne@math.uni-bielefeld.de
    github: LukasKuehne

  - name: Markus Kurtz
    affiliation: University of Kaiserslautern-Landau
    email: kurtz@mathematik.uni-kl.de
    github: mgkurtz

  - name: Benjamin Lorenz
    affiliation: TU Berlin
    website: https://www.tu.berlin/dmg/ueber-uns/systemadministration

>     email: benjamin.lorenz@tu-berlin.de

Reviewer 2: ```suggestion
    email: lorenz@math.tu-berlin.de
```

  - name: Benjamin Schröter  # works on matroids with Lukas Kühne
    affiliation: KTH Stochkholm
    website: https://people.kth.se/~schrot
    email: schrot@kth.se
    github: bschroter

  - name: Justus Springer
    affiliation: Universität Tübingen
    website: https://www.math.uni-tuebingen.de/de/forschung/algebra/personen/justus-springer
    email: justus.springer@uni-tuebingen.de
    github: justus-springer

  - name: Jan Philipp Thiele
    affiliation: WIAS Berlin
    github: jpthiele

  - name: Andrew P. Turner
    affiliation: University of Pennsylvania
    website: https://apturner.net/
    email: turnerap@sas.upenn.edu
    github: apturner

  - name: Grégory Vanuxem
    github: gvanuxem

  - name: Laura Voggesberger
    affiliation: Ruhr-Universität Bochum
    website: https://www.ruhr-uni-bochum.de/ffm/Lehrstuehle/Lehrstuhl-VI/voggesberger.html
    email: laura.voggesberger@ruhr-uni-bochum.de
    github: voggesbe

  - name: Marcel Wack
    affiliation: TU Berlin
    website: https://page.math.tu-berlin.de/~wack/
    email: wack@math.tu-berlin.de
    github: Sequenzer

  - name: Lena Weis
    affiliation: TU Berlin
    website: https://page.math.tu-berlin.de/~weis/

>     email: weis@math.tu-berlin.de

Reviewer 2: ```suggestion
    email: weis@math.tu-berlin.de
    github: lenweis
```

Author: Cool!

I have been trying to find this username and failed. For instance, the one you mention (I take it to be correct) is not listed on the contributors page: https://github.com/oscar-system/Oscar.jl/graphs/contributors. Do you know why?

Author: Or am I just too stupid to see it?

Author: (I am just curious.)

Reviewer 2: > Cool!
> 
> I have been trying to find this username and failed. For instance, the one you mention (I take it to be correct) is not listed on the contributors page: [oscar-system/Oscar.jl/graphs/contributors](https://github.com/oscar-system/Oscar.jl/graphs/contributors). Do you know why?

Maybe because the email that was used in the commits in not listed in the her github settings.
But she is listed in the organization members list: https://github.com/orgs/oscar-system/people?query=lena

    is_active_PI: true

  - name: Claus Fieker
    affiliation: University of Kaiserslautern-Landau
    website: https://math.rptu.de/en/wgs/agag/people/head/fieker
    email: claus.fieker@rptu.de
    github: fieker
    is_active_PI: true

  - name: Max Horn
    affiliation: University of Kaiserslautern-Landau
    website: https://www.quendi.de/math
    email: mhorn@rptu.de
    github: fingolfin
    is_active_PI: true

  - name: Michael Joswig
    affiliation: TU Berlin
    website: https://page.math.tu-berlin.de/~joswig/
    email: joswig@math.tu-berlin.de
    github: micjoswig
    is_active_PI: true

    ######################
    # Active contributors
    ######################

  - name: John Abbott
    affiliation: University of Kaiserslautern-Landau
    website: https://math.rptu.de/ags/agag/personen/mitglieder
    email: john.abbott@rptu.de
    github: JohnAAbbott
    paid_by_dfg: true

  - name: Alex J. Best
    affiliation: King’s College London
    website: https://alexjbest.github.io/
    email: alex.j.best@gmail.com
    github: alexjbest

  - name: Martin Bies
    affiliation: University of Kaiserslautern-Landau
    website: https://martinbies.github.io

>     email: martin.bies@rptu.de

Reviewer: Actually seeing this now: I am not sure we should add email addresses for everyone, this might turn into a privacy / data protection headache, at least if those email addresses are not otherwise public (maybe they all are in the git commit logs, though?)

It should be fine for the PIs. Dunno

Author: All emails are public in at least one way. Say by visiting the linked website and then looking up the email there. Once could (in my case) also go through my github profile, which will again link to my website.

Still, if you feel I should remove emails, then this is of course an easy fix.

For all it matters - I have also added emails for all OSCAR book contributors. So this issue repeats for the OSCAR book website/contributors.

    email: muller@math.uni-sb.de
    github: StevellM

  - name: Giosuè Muratore
    affiliation: Universidade de Lisboa
    website: https://sites.google.com/view/giosue-muratore
    email: muratore.g.e@gmail.com
    github: GiosueMuratore

  - name: Erik Paemurru  # Postdoc with Simon Brandhorst, algebraic geometry
    affiliation: Saarland University
    website: https://www.math.uni-sb.de/ag/brandhorst/index.php?option=com_content&view=article&id=49:paemurru-en&catid=18&lang=en&Itemid=114
    email: paemurru@math.uni-sb.de
    github: paemurru

  - name: Lakshmi Ramesh
    affiliation: University of Kaiserslautern-Landau
    website: https://Lax202.github.io
    email: ramesh@mathematik.uni-kl.de
    github: Lax202

  - name: Yue Ren
    affiliation: Durham University
    website: https://www.yueren.de/
    email: yue.ren2@durham.ac.uk
    github: YueRen

  - name: Morgan Rodgers
    affiliation: University of Kaiserslautern-Landau
    website: https://math.rptu.de/ags/agag/personen/mitglieder
    email: morgan.rodgers@rptu.de
    github: mjrodgers

  - name: Felix Röhrich

>     affiliation: RWTH-Aachen

Reviewer: ```suggestion
    affiliation: RWTH Aachen University
```

## Contributors

The following people are actively contributing to the OSCAR project.

<ul>
{% for p in page.contributors %}
{% if p.is_active_PI != true %}

> {% if p.is_retired != true %}

Reviewer: It might be simpler if you just split the list into three, one for active PIs, one for active contributors, one for former contributors. Then you can remove `is_active_PI` and `is_retired` everywhere

Author: I believe you have a point.

Author: Done.

    {% endif %}
 </li>
{% endif %}
{% endif %}
{% endfor %}
</ul>

## Former Contributors

> The following people did contribute to the OSCAR project in the past, but have not added code since June 2023.

Reviewer: ```suggestion
The following people did contribute to the OSCAR project in the past, but have not contributed code changes in the past 12 month (as of June 2024).
```
otherwise this will be just factually wrong in short order.

Also, I think we should also somewhere place instructions how to update these lists, i.e. mention running `git shortlog -s --since="1 year ago"` in all our repositories. 

And finally we kinda need to remember to update this regularly. Someone should set a regular calendar reminder maybe? Or even partially automate it via a GitHub action (might be too much work for too little gain, though)

Author: I wanted to avoid this issue, hence the hard-coded date. Honestly, I believe we do not want to repeat this exercise once per month or so. Let me ping @aaruni96 so we can gauge if it is feasible to automate the process. If not, I would suggest to keep the hard-coded date. At least for the next month or so.

Reviewer: It would definitely be possible to automate this, but not worth the effort.

But as I see it, your current text does not avoid "the issue", to the contrary: if we don't update the page regularly, then in 6 months it'll suggest that "the following people" did not contribute for the past *18* month. Moreover, if one of them actually makes a contribution next week, we still make this (now incorrect) claim about them.

While with my version, it is clear that (a) this is about "contribution of the past 12 month" (so the next time someone wants to update it, they know what the cut-off point is), and it also makes it clearly that the list is current as of June 2024 -- so someone reading it in December can take this into account "ah, so that's why I am listed under 'former contributor' -- guess I should ping them to update this page"

Author: Sorry my bad. Your suggestion is perfect. I just had to read carefully. Has been applied.

Reviewer 3: I'm sure I can come up with some convoluted way to automatically update this, but its not going to be quickly.

Its trivial to ask for "date one year ago" (`date -d "last year" +"%Y-%m-%d"`), but less clear what to do with the the output of git shortlog, and how to programmatically modify an md file.

Author: Let us gauge in a month or two (presumably someone - maybe I) has to repeat this on a regular (monthly?) basis. If this process turns out to be time consuming each time, then we can investigate further.
